### PR TITLE
Feat/Connect password reset function with frontend

### PR DESCRIPTION
**Pull request by bank1e**
**Purpose:**
- Create UI to navigate user to forgot password page.
- Create UI to initiate password reset function through the URL that user received.

**Change(s):**
- Installed query-string to take token from password reset URL.
- Hard coded local server path in authController. (Reason: It needs the client's local path, not backend local path.)
- Added ForgotPasswordPage and ResetPasswordPage in App.jsx.
- Created new components for Forgot/Reset Password Pages.
- Added link to ForgotPasswordPage in LoginPage.jsx.

**How to review:**
Video_2024-05-31_164658

1. Navigate to click "Forgot Password" link.
2. Input an existing user's email address that you can receive password reset email.
3. If email address exists in database, it sends email to reset password. If not, error "User not found" pops up.
4. Click the URL in the email.
5. Input a new password.
6. Try to log in with the new password you just set, and confirm the success response.

Video_2024-05-31_171923
👆️ Error:  If token is invalid or expired, an error shows up. - _Invalid or expired password reset token_ 

Video_2024-05-31_172127
👆️ Error: Email address does not match. - _User not found_

### CODE REVIEW

**Review by alexh205 — COMMENTED** [review]
- Please add a button to both the Forgot/Reset Password pages to trigger the form submission (as it took me some time initially to initiate the submission). Having visual buttons will provide a more user-friendly functionality to the components. Retain the current trigger and only add the buttons as a secondary trigger. 

- The Reset Password page should have a secondary input field "Verify New Password". This will allow users the ability to confirm the accuracy of their password entry before saving it to the Database. 

Please do let me know if you have any questions.

**Comment by bank1e** [comment]
**Screenshots:**
image
image
image
image

**Changes:**
1.  Added Submit button in forgot/reset password page.
2.  Added minimum length validation in text field in reset password page.
3.  Added password repeat field and validation in reset password page.

**Comment by bank1e** [comment]
Thank you for the comments! Please review the changes again.

**Comment by bank1e** [comment]
image
Added "Repeat password" message in the text field in password reset page.

**Review by alexh205 — COMMENTED** [review]
The updates look great!

Here is one minor update: 
- Currently, within the 'ResetPasswordPage.jsx' module, the 'onSubmit' function will trigger 'fetchResetPasswordAPI' even if the 'newPassword' and 'newPasswordVerify' parameters do not match. If you place the 'fetchResetPasswordApi' function with an 'ELSE' block, then that should fix the issue. 


image

- The two passwords are of different lengths, so the first alert is correct

image
- Once the first alert 'Passwords do not match' is confirmed, the secondary alert 'Password reset successful' is incorrectly triggered. 

image
 - This is the solution to address the incorrect function trigger.


- There also seems to be a merge conflict highlighted blow. Please address that if you can, and if you need help with it, please let me know. 

image


All other functionality works well.

File: client/src/components/ResetPasswordPage/ResetPasswordPage.jsx [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+          "Password reset successful. Please use the new password to log in."
+        );
+      }
+    });
+  };
+  const onSubmit = (event) => {
+    event.preventDefault();
+    if (newPassword !== newPasswordVerify) {
+      alert("Passwords do not match");
+    } else {
```
Comment (bank1e) [inline client/src/components/ResetPasswordPage/ResetPasswordPage.jsx:41]: Thank you for the detailed advise! It is very helpful for correction. 

- As advised, **else** is inserted before fetchResetPasswordApi. 
- Conflict in **App.jsx** has been resolved.

**Review by alexh205 — APPROVED** [review]
Approved

**Pull request merged**